### DISCUSSION TITLE
prefill field form with default locale content when adding translation

### DISCUSSION THREAD

**Discussion by mvdve** [payloadcms/payload#2054] [Feature Requests & Ideas]
Would it be possible to prefill a translatable field with the saved content of the default language? For example:

- Create item in default language
- Switch language to fill translatable fields
- Text field will be filled with the content from the default language which is already saved
- Translate the content, hit save, done!

An option would be to use the defaultValue field option and load the content there, but at the moment only locale and user are populated there. Does somebody has a suggestion on how to handle this without custom frontend components? Handling this on a global scope would be even better.

**Comment by DanRibbens** [comment]
Is this different in some way from the the `fallback` localization option?

I am not sure I follow what you're proposing, if this is different from fallback I'll need you to maybe elaborate more. 

edit - link to docs https://payloadcms.com/docs/configuration/localization#enabling-in-the-payload-config

**Comment by mvdve** [comment]
Hi Dan,

Sorry about that, i'l try to better explain my use case. Admins who translate content within the GUI face an empty form after they switch language after creating a new item via de admin GUI, only the id placeholder for the title is shown.

So the workflow would be:

1. Set admin GUI in default language (NL)
2. Select a collection
3. Fill in the form fields
4. Click save
5. Switch language (EN) to add translation
6. Fill in the form fields
7. Click save to save translation

The question is regarding step 6: Would it be possible to set a default value for the text based form fields, where this default value is de data from the default language. So instead of starting with an empty form, for example the title and a description (EN) will be filled with the title and description from the default language (NL) which makes is easier to translate as you can translate the content directly in the field.

One way to solve this would be to add for example the item data to the defaultValue function arguments, like the validation functionality.

```
// Non working examples
defaultValue: ({ data }) => data.nl.title

// Or
defaultValue: ({ id }) => fetchDataById(id, "nl").title
```

Working with the defaultValue setting is a good option but would be repetitive as it is the same system for all fields. So a field settings could also be an option, something like this:

```
{
  name: 'title',
  type: 'text',
  localized: {
    inputFallback: true
  },
}
```

**Reply by DanRibbens** [reply]
Oh, yeah great explanation!

> defaultValue: ({ data }) => data.nl.title

Yeah that isn't going to work since the UI doesn't fetch the data with `?locale=all`, so you wouldn't have the object structure for locales that you're after.

I like the idea of an option, but there are some larger implications to work through. We currently opt to put the `fallback` as a feature handled by the API that is intentionally obfuscated from the admin UI. One reason for this is that when an editor is working on different languages there would need to be a way to discern and set the value as that of the default locale value or the intended value.

For example if I leave a localized field blank and save a document, the fallback allows the data to still come through in the other locale on the API. If the field went from `undefined` to "The default locale's content" then somebody might assume by updating the default locale that it would update the langauges coming from through as fallback when instead, under your workflow, that would not be the case.

One workaround could be to make a custom component for "Copy from locale" that you put in a UI field into your collections that when clicked (as a button) or selected as a Select, fetches the locale data and sets the form data accordingly. This could be a nice enhancement for editors and in my mind would be more preferred over defaultValue because of the intentionality issue above.

I do think we need to address some of the UX issues around editing multiple locales, as others have pointed out some other shortcomings.
 
https://github.com/payloadcms/payload/discussions/1492
https://github.com/payloadcms/payload/discussions/1308

**Reply by andrejpavlovic** [reply]
I think what would be useful is a way to mark the field as localized or not when editing through the admin. For example, I have a required localized field that I want to localize in only 2 languages, but all other languages (let's say 5 other languages) I want to leave empty so they fallback to default language.

The issue is that if I edit an item in any of the other 5 languages, the field is required and must be filled out. So I cannot actually take advantage of the fallback feature, because I'm required to fill out every language and copy/paste the same data from the default language.

I imagine the workaround currently would be to not make the field required when not editing the default language - though I'm not sure if that's easily doable since required field takes a boolean config value.
